### PR TITLE
New decode_csv_fields processor

### PR DESCRIPTION
**Pull request by adriansr**
This patch introduces a new processor, ~~`decode_csv_field`~~ `decode_csv_fields` that decodes
rows of CSV-formatted data into a string array, one element per column.

```yaml
processors:
- decode_csv_fields:
    fields:
      message: csv
    separator: ,
    overwrite_keys: false
    ignore_missing: false
    trim_leading_space: false
    fail_on_error: true
```

### CODE REVIEW

File: libbeat/processors/actions/decode_csv_field.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	}
+	if _, err = event.PutValue(f.Target, record); err != nil {
+		return event, errors.Wrapf(err, "failed setting field %s", f.Target)
+	}
+	return event, nil
+}
+
+// String returns a string representation of this processor.
+func (f decodeCSVField) String() string {
+	return "decode_csv_field=" + f.Field
```
Comment (andrewkroh) [inline libbeat/processors/actions/decode_csv_field.go]: Would be nice to include more params.

File: libbeat/processors/actions/decode_csv_field_test.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+				Fields: common.MapStr{
+					"message":        `"hello ""world"""`,
+					"existing_field": 42,
+				},
+			},
+			fail: true,
+		},
+
+		"overwrite keys": {
+			config: `
```
Comment (andrewkroh) [inline libbeat/processors/actions/decode_csv_field_test.go]: Personally I would have used `common.MustNewConfigFrom(common.MapStr{...})` to directly define a common.Config and avoid the `unindent` helper.

**Comment by adriansr** [comment]
@andrewkroh 
I originally devised it using the columns mapping too, the problem is that it won't easily fit my current use-case, as I have 50+ columns and need to inspect one of the first columns (a.k.a "type") to decide which mapping to use for the rest.

So I thought maybe it was better to have this processor to decode to an array and then add a generic "extract_array" processor or do this inside an ingest pipeline. I'm planning to discuss this on today's sync with Beats team.

**Comment by andrewkroh** [comment]
This is looking good. I think it just needs a section added to the asciidocs now.

**Comment by adriansr** [comment]
@andrewkroh I've modified the processor a little bit to help align with the rest. Do you mind reviewing again? Now it has docs.

Main change is the rename to *_field**s** and the ability to process more than one field at the same time (I've copied the conf style from your `dns` processor). This adds also the new flag `fail_on_error`, which is also common on another processors.

File: libbeat/docs/processors-using.asciidoc [hunk 3]
```diff
@@ -775,6 +778,51 @@ Adds the environment field to every event:
...
+ifeval::[("{beatname_lc}"=="filebeat") or ("{beatname_lc}"=="journalbeat")]
+* <<decode-csv-fields,`decode_csv_fields`>>
+[[decode-csv-fields]]
+=== Decode CSV fields
+
+experimental[]
+
+The `decode_csv_fields` processor decodes fields containing records in
+comma-separated format (CSV). It will output the values as an array of strings.
+This processor is only available for Filebeat and Journalbeat.
```
Comment (andrewkroh) [inline libbeat/docs/processors-using.asciidoc]: Recommand str
```suggestion
This processor is available for Filebeat and Journalbeat.
```

**Pull request merged**